src/neuroconv/tools/nwb_helpers/_configuration_models/_base_dataset_io.py

```diff
@@ -245,7 +249,9 @@ def model_json_schema(cls, **kwargs) -> dict[str, Any]:
         return super().model_json_schema(mode="validation", schema_generator=PureJSONSchemaGenerator, **kwargs)
 
     @classmethod
-    def from_neurodata_object(cls, neurodata_object: Container, dataset_name: Literal["data", "timestamps"]) -> Self:
+    def from_neurodata_object(
```

Comment:
I would like to make the builder argument optional. If the builder is not passed, then the check for a compound object is not performed.

Reasons:
1. We might find a better way to handle this in the future, and I want to avoid breaking backwards compatibility if that happens.
2. I want a quick way to test the effect of compound types on the full shape.
3. We don't need to complicate tests for cases without compound data types by requiring the building of a manager, passing it, or finding the builder.